messgen/ts_generator.py

```diff
@@ -12,150 +16,166 @@
+        proto_comment = proto.get("comment", "")
+        proto_version = proto.get("version", "")
+
+        output_path = os.path.join(out_dir, proto_name.replace(SEPARATOR, os.sep))
+        os.makedirs(output_path, exist_ok=True)
+        output_file = os.path.join(output_path, f"{proto_name.split(SEPARATOR)[-1]}.ts")
+
+        for type_name in types:
+            self.generate_type(proto_name, type_name, types)
+
+        code = '// === AUTO GENERATED CODE ===\n'
+        if proto_comment:
+            code += f'// {proto_comment}\n'
+        code += f'// Protocol: {proto_name}\n'
+        if proto_version:
+            code += f'// Version: {proto_version}\n'
+        code += '\n'
+
+        if self.imports:
+            code += self.generate_imports(out_dir, output_path)
+
+        code += '\n'.join(self.code_lines)
+
+        with open(output_file, 'w', encoding='utf-8') as f:
+            f.write(code)
+
+
+    def generate_imports(self, out_dir, output_path):
+        imports = ''
+
+        for proto_name, type_name in sorted(self.imports):
+            import_path = os.path.relpath(
+                os.path.join(out_dir, proto_name.replace(SEPARATOR, os.sep), proto_name.split(SEPARATOR)[-1]),
+                output_path
+            ).replace('\\', '/')
+
+            imports += f'import {{ {self.to_camel_case(type_name)} }} from "{import_path}";\n'
+        return imports + '\n'
+    
+    
+    def generate_type(self, proto_name, type_name, types):
+        if (proto_name, type_name) in self.generated_types:
+            return
+        self.generated_types[(proto_name, type_name)] = True
+
+        type_def = types[type_name]
+        type_class = type_def.get("type_class")
+        comment = type_def.get("comment", "")
+        interface_name = self.to_camel_case(type_name)
+
+        if type_class == "struct":
+            fields = type_def.get("fields") or []
+            for field in fields:
+                field_type = field["type"]
+                base_type = self.get_base_type(field_type)
+                if not self.is_builtin_type(base_type):
+                    self.handle_custom_type(base_type, proto_name)
+        elif type_class == "enum":
+            pass
+        else:
+            pass  
+
+        if comment:
+            self.code_lines.append(f"/** {comment} */")
+
+        if type_class == "struct":
+            self.code_lines.append(f"export interface {interface_name} {{")
+            fields = type_def.get("fields") or []
+            for field in fields:
+                field_comment = field.get("comment", "")
+                field_name = field["name"]
+                field_type = field["type"]
+                ts_type = self.get_ts_type(field_type, proto_name)
+                if field_comment:
+                    self.code_lines.append(f"  /** {field_comment} */")
+                self.code_lines.append(f"  {field_name}: {ts_type};")
+            self.code_lines.append("}\n")
```

Comment:
AFAIK, the convention for adding new lines is to use `append("")` like here: https://github.com/microavia/messgen/blob/v1-dispatch/messgen/cpp_generator.py#L126

```suggestion
            self.code_lines.append("}")
            self.code_lines.append("")
```